flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesResourceManager.java

```diff
@@ -338,6 +339,7 @@ protected FlinkKubeClient createFlinkKubeClient() {
 
 	@Override
 	protected double getCpuCores(Configuration configuration) {
-		return flinkConfig.getDouble(KubernetesConfigOptions.TASK_MANAGER_CPU, numSlotsPerTaskManager);
+		double fallback = configuration.getDouble(KubernetesConfigOptions.TASK_MANAGER_CPU);
+		return TaskExecutorProcessUtils.getCpuCoresWithFallback(configuration, fallback).getValue().doubleValue();
```

Comment:
Since K8s/Yarn/Mesos share the same logics to get cpu cores, maybe we could provide a unified method. Then we will not need to test for each implementation.
```
public static <T> CPUResource getCpuCoresWithFallback(final Configuration config, ConfigOption<T> fallback)
```

Reply:
I think this is difficult to unify among all active RMs, because we do operations over concrete number types in `getCpuCoresWithFallback` and yarn fallback option is Integer but others are double.
We could have one more shortcut for double though for Mesos/K8s:
```
public static CPUResource getCpuCoresWithFallback(final Configuration config, ConfigOption<Double> fallback)
```

Reply:
I agree with @azagrebin that it is difficult to unify Yarn with K8s/Mesos. We cannot easily get the type of the config option, which means it's hard to properly read the config option into a double value.

I'll create a util method share by K8s/Mesos.

Regarding testing, I think we need this test for each deployment anyway, to test the proper fallback of different config options on different deployment.